pkg/scyllaclient/client_rclone.go

```diff
@@ -416,8 +417,9 @@ func (c *Client) RcloneListDir(ctx context.Context, host, remotePath string, opt
 // The remotePath is given in the following format "provider:bucket/path".
 // Resulting item path is relative to the remote path.
 func (c *Client) RcloneListDirIter(ctx context.Context, host, remotePath string, opts *RcloneListDirOpts, f func(item *RcloneListDirItem)) error {
-	ctx = customTimeout(ctx, c.config.ListTimeout)
 	ctx = noRetry(ctx)
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I guess we need to explicitly disable the timeout.

Reply:
Not sure what you mean. If this context comes from a caller with a different timeout attached shouldn't we still respect that?

Reply:
Sure. Check [this](https://github.com/scylladb/scylla-manager/blob/master/pkg/scyllaclient/timeout.go) atm there is no need to send something with no timeout for the whole operation.

Try changing server timeout parameter in the test.

Reply:
Ah, now I understand. As it is now there doesn't appear to be a way to disable it. We could either give it a very long value for the custom timeout within this Iter function the way it did the 12h before, or can modify the timeout function so it considers a duration of 0 to be no timeout.

Reply:
Let's add a separate function noTimeout, it's more explicit and it's easier to track it's users.